### PR TITLE
[KYUUBI #4439][FOLLOWUP] Add dto class for operation data

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

- add dto for operation data, which is programming friendly
- show exception for session data

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by turboFei** [comment]
also cc @lightning-L

**Comment by turboFei** [comment]
BTW, we can add some filter conditions for the list sessions/operations api.

Such as filter the sessions/operations with user list.

File: kyuubi-rest-client/src/main/java/org/apache/kyuubi/client/api/v1/dto/OperationData.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.client.api.v1.dto;
+
+import java.util.Objects;
+import org.apache.commons.lang3.builder.ReflectionToStringBuilder;
+import org.apache.commons.lang3.builder.ToStringStyle;
+
+public class OperationData {
```
Comment (pan3793) [inline kyuubi-rest-client/src/main/java/org/apache/kyuubi/client/api/v1/dto/OperationData.java:24]: should we add a conf field since that `ExecuteStatement` supports `confOverlay`?

File: kyuubi-common/src/main/scala/org/apache/kyuubi/Utils.scala [hunk 2]
```diff
@@ -387,4 +387,19 @@ object Utils extends Logging {
     Option(Thread.currentThread().getContextClassLoader).getOrElse(getKyuubiClassLoader)
 
   def isOnK8s: Boolean = Files.exists(Paths.get("/var/run/secrets/kubernetes.io"))
+
+  /**
+   * Return a nice string representation of the exception. It will call "printStackTrace" to
+   * recursively generate the stack trace including the exception and its causes.
+   */
+  def exceptionString(e: Throwable): String = {
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/Utils.scala:395]: prettyPrint

File: kyuubi-rest-client/src/main/java/org/apache/kyuubi/client/api/v1/dto/OperationData.java (hunk 1 above)
Comment (turboFei) [reply kyuubi-rest-client/src/main/java/org/apache/kyuubi/client/api/v1/dto/OperationData.java:24]: difficult to add

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Is https://kyuubi.readthedocs.io/en/master/client/rest/rest_api.html outdated?

**Comment by turboFei** [comment]
thanks, merged to master
